src/macaron/vsa/vsa.py

```diff
@@ -135,11 +144,42 @@ class VerificationResult(StrEnum):
     PASSED = "PASSED"
 
 
+def get_common_purl_from_artifact_purls(purl_strs: Iterable[str]) -> str | None:
+    """Get a single common PackageURL given some artifact PackageURLs.
+
+    Assumption: A package may have more than one artifact. If each artifact is identified
+    by a PackageURL, these PackageURLs still share the type, namespace, name, and
+    version values. The common PackageURL contains these values.
+    """
+    try:
+        purls = [PackageURL.from_string(_) for _ in purl_strs]
```

Comment:
Interesting, I didn't know that using `_` like this could work! 
nit: I think it  would be clearer to use a variable name instead.

Reply:
See 6f948e6a56d897cdd91999ce3c847c0eb021a539.

Besides the point, but I initially coded this as `list(map(PackageURL.from_string, purl_strs))`, then changed to this since list comprehension, since is usually regarded as being more idiomatic in Python. The `_` was something I learned from reading some Scala code a while back, and then trying it in Python.